### PR TITLE
Fixed several sync bugs based on load test.

### PR DESCRIPTION
**Pull request by stephanheilner**

### CODE REVIEW

**Comment by stephanheilner** [comment]
The 64-bit Int.max was causing sync to fail because it can't handle such a large number.

File: LDSAnnotations/SyncAnnotationsOperation.swift [hunk 1]
```diff
@@ -50,6 +50,7 @@ class SyncAnnotationsOperation: Operation {
     var deserializationErrors = [NSError]()
     
     private let source: NotificationSource = .Sync
+    private let awsMaxInt = 99999
```
Comment (hiltonc) [inline LDSAnnotations/SyncAnnotationsOperation.swift]: To me AWS == Amazon Web Services. Can you call it sync service or something like that?

File: LDSAnnotations/AnnotationStore+Bookmark.swift [hunk 2]
```diff
@@ -132,11 +132,11 @@ extension AnnotationStore {
             // Mark all the annotations that were updated as changed
             try annotationIDs.forEach { try self.updateLastModifiedDate(annotationID: $0, source: source) }
             
-            return try self.addBookmark(name: name, paragraphAID: paragraphAID, displayOrder: displayOrder, annotationID: annotation.id, source: source)
+            return try self.addBookmark(name: name, paragraphAID: paragraphAID, displayOrder: displayOrder, annotationID: annotation.id, offset: Bookmark.DisplayOrderMax, source: source)
```
Comment (hiltonc) [inline LDSAnnotations/AnnotationStore+Bookmark.swift]: This seems like a mismatch. Why are we passing `DisplayOrderMax` to `offset`?

File: LDSAnnotations/Bookmark.swift [hunk 3]
```diff
@@ -65,7 +66,7 @@ public struct Bookmark: Equatable {
         if let paragraphAID = paragraphAID {
             result["@pid"] = paragraphAID
         }
-        if let displayOrder = displayOrder {
+        if let displayOrder = displayOrder where displayOrder != Bookmark.DisplayOrderMax {
```
Comment (hiltonc) [inline LDSAnnotations/Bookmark.swift]: Why does `displayOrder` take two special values (`nil` and `DisplayOrderMax`), can't it just use `nil` to represent "no display order"?

Comment (stephanheilner) [inline LDSAnnotations/Bookmark.swift]: `.order(` supports using `??`, so I'm going to follow your suggestion and use `?? Int.max`, which should put the nulls at the end.

File: LDSAnnotations/SyncNotebooksOperation.swift [hunk 4]
```diff
@@ -178,6 +177,16 @@ class SyncNotebooksOperation: Operation {
                                 downloadedNotebook = try self.annotationStore.addNotebook(uniqueID: uniqueID, name: name, description: description, status: status, lastModified: lastModified, source: self.source)
                             }
                             downloadedNotebooks.append(downloadedNotebook)
+                            
+                            if let annotationUniqueIDs = annotationUniqueIDs where !annotationUniqueIDs.isEmpty {
```
Comment (hiltonc) [inline LDSAnnotations/SyncNotebooksOperation.swift]: This can be simplified to:

    `for annotationUniqueID in annotationUniqueIDs ?? [] {`

**Comment by stephanheilner** [comment]
I'm not 100% sure it works to be honest, but I was going through the source to see how to create an `IFNULL` Expression, and then found they have a function for `??` for the IFNULL function in SQLite.

**Comment by stephanheilner** [comment]
Okay to merge?

**Comment by hiltonc** [comment]
There's the one last recommendation if you want to do it. 👍 otherwise.

**Comment by stephanheilner** [comment]
Oh right, thought I'd already pushed that change.

**Pull request merged**